Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,287 @@
+            seedChem, plantHarvestType, exudeGases, plantMutations, false, plantSpeciation, tolerances.ToString(), generalTraits.ToString(), _scanMode);
+    }
+
+    /// <summary>
+    ///     Returns information about the seed.
+    /// </summary>
+    public string CheckGeneralTraits(SeedData plant, string generalTraits)
+    {
+        var name = Loc.GetString(plant.DisplayName);
+        var endurance = plant.Endurance;
+        var yield = plant.Yield;
+        var lifespan = plant.Lifespan;
+        var maturation = plant.Maturation;
+        var growthStages = plant.GrowthStages;
+        var potency = plant.Potency;
+
+        generalTraits = name + ";" + endurance + ";" + yield.ToString() + ";" + lifespan + ";" + maturation + ";" + growthStages + ";" + potency;
+
+        return generalTraits;
+    }
+    public string CheckAllMutations(SeedData plant, string plantMutations)
+    {
+        if (plant.TurnIntoKudzu) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-turnintokudzu")}";
+        if (plant.Seedless) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-seedless")}";
+        if (plant.Slip) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-slip")}";
+        if (plant.Sentient) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-sentient")}";
+        if (plant.Ligneous) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-ligneous")}";
+        if (plant.Bioluminescent) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-bioluminescent")}";
+        if (plant.CanScream) plantMutations += "\r\n   " + $"{Loc.GetString("plant-analyzer-mutation-canscream")}";
+
+        return plantMutations;
+    }
+    public string CheckAllTolerances(SeedData plant, string tolerances)
+    {
+        var nutrientConsumption = plant.NutrientConsumption;
+        var waterConsumption = plant.NutrientConsumption;
+        var idealHeat = plant.IdealHeat;
+        var heatTolerance = plant.HeatTolerance;
+        var idealLight = plant.IdealLight;
+        var lightTolerance = plant.LightTolerance;
+        var toxinsTolerance = plant.ToxinsTolerance;
+        var lowPresssureTolerance = plant.LowPressureTolerance;
+        var highPressureTolerance = plant.HighPressureTolerance;
+        var pestTolerance = plant.PestTolerance;
+        var weedTolerance = plant.WeedTolerance;
+
+        tolerances = nutrientConsumption + ";" + waterConsumption + ";" + idealHeat + ";" + heatTolerance + ";" + idealLight + ";" + lightTolerance + ";"
+                   + toxinsTolerance + ";" + lowPresssureTolerance + ";" + highPressureTolerance + ";" + pestTolerance + ";" + weedTolerance;
+
+        return tolerances;
+    }
+    private void OnModeSelected(EntityUid uid, PlantAnalyzerComponent component, PlantAnalyzerSetMode args)
+    {
+        SetMode(uid, component, args.AdvancedScan);
+    }
```

Comment:
whitespace

Reply:
I really can't figure out which whitespace you mean here?